### PR TITLE
[TASK] Remove option --force for FlushCachesTask

### PR DESCRIPTION
**Pull request by sabbelasichon**
* **Please check if the PR fulfills these requirements**
- [x] Tests for the changes have been added (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Bug fix

* **What is the current behavior?** (You can also link to an open issue here)
The option --force for task FlushCachesTask is uses. This was deprecated a long time ago and is now removed in version > 6.2

* **What is the new behavior (if this is a feature change)?**
Remove the option --force completely. 

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
Yes. You cannot use Surf with older Versions of typo3_console and you should not.

### CODE REVIEW

File: src/Task/TYPO3/CMS/CompareDatabaseTask.php [hunk 1]
```diff
@@ -33,6 +33,7 @@ class CompareDatabaseTask extends AbstractCliTask
 {
     public function execute(Node $node, Application $application, Deployment $deployment, array $options = [])
     {
+        /** @var $application CMS */
```
Comment (simonschaufi) [inline src/Task/TYPO3/CMS/CompareDatabaseTask.php:36]: first type and then variable name: `/** @var CMS $application */`

File: src/Task/TYPO3/CMS/FlushCachesTask.php [hunk 2]
```diff
@@ -22,7 +23,12 @@ class FlushCachesTask extends AbstractCliTask
 {
     public function execute(Node $node, Application $application, Deployment $deployment, array $options = [])
     {
+
+        /** @var $application CMS */
```
Comment (simonschaufi) [inline src/Task/TYPO3/CMS/FlushCachesTask.php:27]: same here

File: src/Task/TYPO3/CMS/FlushCachesTask.php [hunk 3]
```diff
@@ -37,13 +43,22 @@ public function execute(Node $node, Application $application, Deployment $deploy
...
+                }
+                return [$this->getConsoleScriptFileName($node, $application, $deployment, $options), 'cache:flush'];
             default:
                 return [];
         }
     }
+
+    protected function resolveOptions(OptionsResolver $resolver)
+    {
+        $resolver->setDefault('flushCacheOptions', ['--files-only']);
```
Comment (t3easy) [inline src/Task/TYPO3/CMS/FlushCachesTask.php:61]: Should `--files-only` be the dafault?

**Comment by simonschaufi** [comment]
is this a breaking change for existing scripts?

**Comment by t3easy** [comment]
> is this a breaking change for existing scripts?
Only if someone would use a Surf 3.x with an old project using typo3-console 5.
But we could add a check later.

File: src/Task/TYPO3/CMS/FlushCachesTask.php (hunk 3 above)
Comment (simonschaufi) [reply src/Task/TYPO3/CMS/FlushCachesTask.php]: are you waiting for an answer here? I first need to know what is the diff between the 2 options.

Comment (t3easy) [reply src/Task/TYPO3/CMS/FlushCachesTask.php]: https://docs.typo3.org/p/helhum/typo3-console/6.3/en-us/CommandReference/CacheFlush.html
https://github.com/TYPO3-Console/TYPO3-Console/blob/e409b443e00673390e42190ed706ebd59043de2b/Classes/Console/Command/Cache/CacheFlushCommand.php#L57-L64
https://github.com/TYPO3-Console/TYPO3-Console/blob/e409b443e00673390e42190ed706ebd59043de2b/Classes/Console/Service/CacheLowLevelCleaner.php#L30-L35

TYPO3 Console only deletes the content of the cache folder. No database or redis caches are flushed.
As Surf by default has an empty cache folder when we switch the current release, it wouldn't do enything.

Comment (simonschaufi) [reply src/Task/TYPO3/CMS/FlushCachesTask.php]: So do we want to clear all caches by default? When we use the files-only, it doesn't do anything?

Comment (t3easy) [reply src/Task/TYPO3/CMS/FlushCachesTask.php]: Surf always flushed all caches in the past.

If someone needs to flush only files, the option exists.
I would use cache:flush without a parameter by default.

Comment (simonschaufi) [reply src/Task/TYPO3/CMS/FlushCachesTask.php]: agree. flush all caches by default.

**Comment by YetiCGN** [comment]
Could you backport this to the 2.0 branch? I run 2.2.1 finally after years on the master branch and I would really like to keep doing that.

**Comment by t3easy** [comment]
https://github.com/TYPO3/Surf/pull/411

**Comment by YetiCGN** [comment]
@t3easy Thanks! So I take it, that option is not removed for 2.x but overridable. Is there an example of how to get down to the options of the FlushCachesTask?

**Comment by t3easy** [comment]
No, use something like this:

```
$application->setOption(TYPO3\Surf\Task\TYPO3\CMS\FlushCachesTask::class . '[arguments]', [])
```
See

https://github.com/TYPO3/Surf/blob/2.0/Documentation/Examples/TYPO3/Index.rst

https://github.com/TYPO3/Surf/commit/dc0c856126d559d602f0569d11ccf359c6e95fb6

**Pull request merged**